Fix multiple issues with FinalizationRegistry content (#17126)
mdn/content: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md

### Parameters

- `unregisterToken`
  - : The token used with the {{jsxref("FinalizationRegistry.prototype.register", "register")}} method when registering the target object. Multiple cells registered with the same `unregisterToken` will be unregistered together.

### Return value

> `true` if at least one cell was unregistered; `false` if no cell was unregistered.

Reviewer: ```suggestion
A boolean value that is `true` if at least one cell was unregistered, `false` if no cell was unregistered.
```

> `true` if at least one cell was unregistered; `false` if no cell was unregistered.

Reviewer: In addition to the boolean value, I think there are cases where `undefined` can be returned, am I wrong?

Author: [Can't find it in spec](https://tc39.es/ecma262/#sec-finalization-registry.prototype.unregister). It either returns a boolean, or throws a `TypeError`.

> 

Reviewer: I think we have cases where the `{{jsxref("TypeError")}}` exception is raised. We likely want an `### Exceptions` section.

Author: Makes sense—when I work on these docs there are a lot of sections missing mentions of exceptions. Is there a convention for how this should be structured? Or just an h3 section.

Reviewer: Yes, we should add them as we do for API, with a definition list in that section: https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_method_subpage_template#exceptions

Something like: https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/continue#exceptions

And yes, there are a lot of exceptions not listed (both under JS and API), I would like to make a full check once, but I don't have the time. Getting the exception list, with causes, complete under JavaScript/ and API/ is hard, though as they are buried inside the prose, or the algorithm, of the specifications.

### Return value

A boolean value that is `true` if at least one cell was unregistered and `false` if no cell was unregistered.

### Exceptions

> A `TypeError` is thrown if `unregisterToken` is not an object.

Reviewer: ```suggestion
- {{jsxref("TypeError")}}
  - : Thrown when `unregisterToken` is not an object.
```
